system/jhtree/jhtree.cpp

```diff
@@ -1762,9 +1769,26 @@ bool CKeyCursor::_next(IContextLogger *ctx)
                 if (node != NULL)
                 {
                     nodeKey = 0;
+                    unsigned depth = key.getBranchDepth();
+                    while (depth)
+                    {
+                        depth--;
+                        if (depth < maxParentNodes && parents[depth])
+                        {
+                            unsigned numParentKeys = parents[depth]->getNumKeys();
+                            unsigned parentNodeKey = parentNodeKeys[depth];
+                            if (parentNodeKey < numParentKeys-1)
+                            {
+                                parentNodeKeys[depth] = parentNodeKey;
```

Comment:
I think this should be parentNodeKey+1.

Reply:
I think you are right. Would be a no-op without the +1